src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -651,10 +654,50 @@ private static string GenerateClassName(TypeDeclarationSyntax typeDeclaration)
                                 return (null, true);
                             }
                         }
+
+                        shadowedNames.Add(fs.Name);
```

Comment:
>shadowedNames [](http://example.com/codeflow?start=24&length=13)

Could you please add a comment explaining what this shadowedNames for? You may put your example you mentioned before in the comment. Thanks!

Reply:
Sure thing!